src/components/seo.jsx

```diff
@@ -1,10 +1,15 @@
 import Head from "next/head";
 
 export default function SEO({ title, description, imageUrl, url }) {
-	const canonicalUrl = url
-		? // eslint-disable-next-line no-restricted-globals, n/prefer-global/process, n/prefer-global/url
-			new URL(url, process.env.NEXT_PUBLIC_SITE_URL)
-		: undefined;
+	let canonicalUrl = false;
+
+	if (url) {
+		// eslint-disable-next-line no-restricted-globals, n/prefer-global/process, n/prefer-global/url
+		const urlObject = new URL(url, process.env.NEXT_PUBLIC_SITE_URL);
+		// Remove any query string as search is writable so we remove any potential query string
+		urlObject.search = "";
```

Comment:
This fixes the search string issues but I could imagine others potentially. I'm wondering if this is the correct fix or if the issue is the URL we are passing in is from the router and not from WP.

Reply:
@moonmeister Ah I see your point. Let me fix this. Thanks for spotting this ❤️

Reply:
@moonmeister Sorry I might be missing something here.

But if you have a look at the blog post slug template we have 

`const { title, date, author, uri, excerpt, editorBlocks } = post;`

and that uri is used for the SEO component

```
			<Seo
				title={title}
				url={uri}
				description={excerpt.replaceAll(/<\/?\S+>/gm, "")}
			/>
```

So it looks to me we are using the WP URL and sorry if I am totally missing something.

Reply:
FWIW, this scenario cleans things: 

![image](https://github.com/user-attachments/assets/2a27288d-364c-493a-8886-802f7741c906)

Reply:
I just updated the original ticket with a video. TL;DR: This only affects doc pages as supported by my testing and the analytics page I linked to there.

Reply:
Thanks @moonmeister 🚀 

@alexwpengine is going to have a look. (Thanks @alexwpengine )